lib/src/domain/usecase/usecase_interface.dart

```diff
@@ -0,0 +1,11 @@
+import '../../core/state/state.dart';
+import '../entity/genre.dart';
+import '../entity/movie.dart';
+
+abstract class IUseCaseMovie {
+  Future<DataState<List<Movie>>> repositoryCall();
+}
+
+abstract class IUseCaseGenres {
+  Future<DataState<List<Genre>>> repositoryCall();
+}
```

Comment:
let's do an interface with generics so we can avoid create an interface for each type of return

Reply:
Done